streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -992,6 +995,16 @@ public void setStateListener(final StreamThread.StateListener listener) {
     }
 
     /**
+     * Set the listener invoked at the beginning, end of batch updates and the conclusion of
+     * restoring a {@link StateStore}.
+     *
+     * @param stateRestoreListener  listener for capturing state store restoration status.
+     */
+    public void setStateRestoreListener(final StateRestoreListener stateRestoreListener) {
```

Comment:
Don't think so, the `stateRestoreListener` set by user is passed through to the `StreamThread`.  Javadoc in `StateRestoreListener` states that it expects operations to be stateless.